packages/types/src/schemas/common.ts

```diff
@@ -203,13 +203,18 @@ export const editorSchema = z.union([
   }),
 ]);
 
+const customSchema = z.record(z.string(), z.any());
+
 export type TerminalPanelType = z.infer<typeof panelTypeSchema>;
 export type TerminalSchema = z.infer<typeof terminalSchema>;
 export type EditorSchema = z.infer<typeof editorSchema>;
+export type CustomSchema = z.infer<typeof customSchema>;
 
 export const webcontainerSchema = commandsSchema.extend({
   meta: metaTagsSchema.optional(),
 
+  custom: customSchema.optional(),
```

Comment:
Oh I think we could add a `.describe` to explain what this is for (it shows up in the extension after).